ngraph/core/src/function.cpp

```diff
@@ -347,6 +370,38 @@ bool Function::visit_attributes(AttributeVisitor& visitor)
     return true;
 }
 
+void Function::add_sinks(const SinkVector& sinks)
+{
+    m_sinks.insert(m_sinks.begin(), sinks.begin(), sinks.end());
+    validate_nodes_and_infer_types();
+}
+
+void Function::remove_sink(const std::shared_ptr<op::Sink>& sink)
+{
+    string remove_name = sink->get_name();
+    m_sinks.erase(std::remove_if(m_sinks.begin(),
+                                 m_sinks.end(),
+                                 [remove_name](std::shared_ptr<op::Sink>& s) {
+                                     return s->get_name() == remove_name;
```

Comment:
compare by name?? not sure about it. It's better to compare sinks
@ilyachur, please verify

Reply:
I propose to use pointers instead of names.

Reply:
fixed